proto/eraftpb.proto

```diff
@@ -80,6 +81,8 @@ message HardState {
     uint64 term = 1;
     uint64 vote = 2;
     uint64 commit = 3;
+    // If the peer goes down we need to be aware of a partially completed membership change.
+    ConfChange pending_membership_change = 4;
 }
 
 message ConfState {
```

Comment:
How about 
```protobuf
struct ConfState {
    repeated uint64 nodes = 1;
    repeated uint64 learners = 2;
    repeated uint64 conf_new_nodes = 3;
    repeated uint64 conf_new_learners = 4;
    int membership_change_start_at = 5;
}
```
And we don't need to store `ConfChange`. We can construct it from fields in `ConfState`.

Reply:
@hicqu and I spoke offline (In person, even!) about this and we think it might be more clear to think of Joint Consensus as moving **between** `ConfState`s, so instead we will try having a `pending_conf_state` stored alongside `conf_state` in the `RaftState` and the Snapshots.